.github/workflows/release.yml

```diff
@@ -1,16 +1,63 @@
-name: Release on Maven Central and GitHub
+name: Release
 
 on:
   workflow_dispatch:
     inputs:
-      deploy-maven-central:
-        description: "Deploy to Maven Central"
+      skip-deploy-maven-central:
+        description: "Skip deployment to Maven Central"
         required: true
         type: boolean
         default: false
 
 jobs:
   release:
     runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: "bash"
+    concurrency:
+      group: ${{ github.workflow }}
+      cancel-in-progress: false
+    permissions:
+      contents: write # Required for creating GitHub release
     steps:
-      - run: echo "Dummy workflow for testing"
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Fail if not running on main branch
+        if: ${{ github.ref != 'refs/heads/main' }}
+        uses: actions/github-script@v7
+        with:
+          script: |
+            core.setFailed('Not running on main branch, github.ref is ${{ github.ref }}. Please start this workflow only on main')
+
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v4
+        with:
+          distribution: "temurin"
+          java-version: 17
+          cache: "maven"
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.OSSRH_GPG_SECRET_KEY }}
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE
+
+      - name: Build
+        run: mvn --batch-mode -T 1C clean install
+
+      - name: List secret GPG keys
+        run: gpg --list-secret-keys
```

Comment:
This ends up in a log. If you need it for debugging, obfuscate the results (e.g. only show 3 digits of the key ID).

Reply:
This outputs information that is public anyway: https://keyserver.ubuntu.com/pks/lookup?search=code%40itsallcode.org&fingerprint=on&op=index

Reply:
:heavy_check_mark: